src/rusoto_driver.rs

```diff
@@ -118,7 +118,11 @@ fn rusoto_region(env: &Environment) -> Result<Region> {
             name: Region::default().name().to_owned(),
             endpoint: e.to_owned(),
         },
-        (None, None) => Region::default(),
+        (None, None) => {
+            let region = Region::default();
+            env.set_region(region.name().to_string(), Setter::Environment);
+            region
+        }
     };
```

Comment:
I think this method should just return the region, and runner should then set it on the env. That will remove all the mut stuff.